### PR TITLE
PPTP-1128 - Returns Display

### PR DESCRIPTION
**Pull request by kiranprakashdigital**
### Description of Work carried through

Brief description of what/why/how.

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/connectors/models/eis/returns/Return.scala [hunk 1]
```diff
@@ -41,13 +41,14 @@ object ExportChargeDetails {
...
 }
 
-case class ReturnsSubmissionResponse(
+case class Return(
   processingDate: String,
   idDetails: IdDetails,
   chargeDetails: Option[ChargeDetails],
-  exportChargeDetails: Option[ExportChargeDetails]
+  exportChargeDetails: Option[ExportChargeDetails],
+  returnDetails: Option[EisReturnDetails]
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtaxreturns/connectors/models/eis/returns/Return.scala:49]: According to the API specs the returns submission response and the get return response are not quite the same but I agree that it makes sense to use a unified representation here. Good call.

File: test/uk/gov/hmrc/plasticpackagingtaxreturns/connectors/ReturnsConnectorISpec.scala [hunk 2]
```diff
@@ -71,24 +72,82 @@ class ReturnsConnectorISpec extends ConnectorISpec with Injector with Subscripti
...
+                              totalWeight = BigDecimal(16466.88),
+                              taxDue = BigDecimal(4600)
+             )
+           )
+    )
+
+  private def stubSuccessfulReturnsSubmission(returnId: String, resp: Return) =
     stubFor(
-      put(s"/plastic-packaging-tax/returns/PPT/$returnId")
+      put(urlMatching(s"/plastic-packaging-tax/returns/PPT/$returnId"))
```
Comment (RyanKellyHMRC) [inline test/uk/gov/hmrc/plasticpackagingtaxreturns/connectors/ReturnsConnectorISpec.scala:150]: I think you only need to use `urlMatching` when you want a regex match.

**Review by RyanKellyHMRC — APPROVED** [review]
One or two very minor suggestions.

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/connectors/ReturnsConnector.scala [hunk 3]
```diff
@@ -79,4 +77,35 @@ class ReturnsConnector @Inject() (httpClient: HttpClient, override val appConfig
...
+            s"  and response has submissionId [${response.idDetails.submissionId}]"
+        )
+        Right(response)
+      }
+      .recover {
+        case httpEx: UpstreamErrorResponse =>
+          logger.warn(
+            s"Upstream error returned on fetching return with correlationId [${correlationIdHeader._2}] and " +
+              s"pptReference [$pptReference], status: ${httpEx.statusCode}, body: ${httpEx.getMessage()}"
+          )
```
[the remark this answers is not in the document]
Comment (kiranprakashdigital) [reply app/uk/gov/hmrc/plasticpackagingtaxreturns/connectors/ReturnsConnector.scala:100]: I get the submissionId only in response, so when it throws an exception there is no submissionId to log. I will probably add the periodKey

File: test/uk/gov/hmrc/plasticpackagingtaxreturns/connectors/ReturnsConnectorISpec.scala (hunk 2 above)
Comment (kiranprakashdigital) [reply test/uk/gov/hmrc/plasticpackagingtaxreturns/connectors/ReturnsConnectorISpec.scala:150]: Yeah as I am using stringInterpolator.. i used urlMatching

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/connectors/ReturnsConnector.scala [hunk 4]
```diff
@@ -79,4 +77,35 @@ class ReturnsConnector @Inject() (httpClient: HttpClient, override val appConfig
...
+    val correlationIdHeader: (String, String) = correlationIdHeaderName -> UUID.randomUUID().toString
+
+    httpClient.GET[Return](appConfig.returnDisplayUrl(pptReference, periodKey),
+                           headers = headers :+ correlationIdHeader
+    )
+      .andThen { case _ => timer.stop() }
+      .map { response =>
+        logger.info(
+          s"Retrieved return display with correlationId [$correlationIdHeader._2], pptReference [$pptReference]" +
+            s"  and response has submissionId [${response.idDetails.submissionId}]"
```
[the remark this answers is not in the document]
Comment (kiranprakashdigital) [reply app/uk/gov/hmrc/plasticpackagingtaxreturns/connectors/ReturnsConnector.scala:91]: Fixed

**Pull request merged**